Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -251,5 +263,516 @@ namespace AZ
+                {
+                    const float offset = (nextFrameBoundaryPixel - lastFrameBoundaryPixel - labelWidth) /2;
+                    const float textBeginPixel = lastFrameBoundaryPixel + offset;
+                    const float textEndPixel = textBeginPixel + labelWidth;
+
+                    // Execution time label
+                    drawList->AddText({ textBeginPixel, wy + ImGui::GetWindowHeight() / 4 }, IM_COL32_WHITE, label.c_str());
+
+                    // Left side
+                    drawList->AddLine(
+                        { lastFrameBoundaryPixel, wy + ImGui::GetWindowHeight() / 2 },
+                        { textBeginPixel - 5, wy + ImGui::GetWindowHeight() / 2},
+                        IM_COL32_WHITE);
+
+                    // Right side
+                    drawList->AddLine(
+                        { textEndPixel, wy + ImGui::GetWindowHeight()/2 },
+                        { nextFrameBoundaryPixel,  wy + ImGui::GetWindowHeight()/2 },
+                        IM_COL32_WHITE);
+                }
+                else // Cannot fit inside, just draw a line between the two boundaries
+                {
+                    drawList->AddLine(
+                        { lastFrameBoundaryPixel, wy + ImGui::GetWindowHeight() / 2 },
+                        { nextFrameBoundaryPixel, wy + ImGui::GetWindowHeight() / 2 },
+                        IM_COL32_WHITE);
+                }
+
+                // Left bound
+                drawList->AddLine(
+                    { lastFrameBoundaryPixel, wy },
+                    { lastFrameBoundaryPixel, wy + ImGui::GetWindowHeight() },
+                    IM_COL32_WHITE);
+
+                // Right bound
+                drawList->AddLine(
+                    { nextFrameBoundaryPixel, wy },
+                    { nextFrameBoundaryPixel, wy + ImGui::GetWindowHeight() },
+                    IM_COL32_WHITE);
+
+                lastFrameBoundaryItr = nextFrameBoundaryItr;
+                nextFrameBoundaryItr++;
+            }
+        }
+
+        inline AZStd::sys_time_t ImGuiCpuProfiler::GetViewportTickWidth() const
+        {
+            return m_viewportEndTick - m_viewportStartTick;
+        }
+
+        inline float ImGuiCpuProfiler::ConvertTickToPixelSpace(AZStd::sys_time_t tick) const
+        {
+            const float wx = ImGui::GetWindowPos().x;
+            const float tickSpaceShifted = aznumeric_cast<float>(tick - m_viewportStartTick); // This will be close to zero, so FP inaccuracy should not be too bad
+            const float tickSpaceNormalized = tickSpaceShifted / GetViewportTickWidth();
+            const float pixelSpace = tickSpaceNormalized * ImGui::GetWindowWidth() + wx;
+            return pixelSpace;
+        }
+
+        // Tick bus overrides
+        inline void ImGuiCpuProfiler::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] ScriptTimePoint time)
+        {
+            m_frameEndTicks.push_back(AZStd::GetTimeNowTicks());
```

Comment:
The tick bus may not be reliable to map each OnTick to one rendering frame. 
OnSystemTick() from SystemTickBus is synced with RPISystem render tick which I think it's better to use that.